cmd/goal/wallet.go

```diff
@@ -200,6 +201,53 @@ var listWalletsCmd = &cobra.Command{
 	},
 }
 
+var renameWalletCmd = &cobra.Command{
+	Use:   "rename [wallet name] [new wallet name]",
+	Short: "Rename wallet",
+	Args:  cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		dataDir := datadir.EnsureSingleDataDir()
+
+		client := ensureKmdClient(dataDir)
+
+		walletName := []byte(args[0])
+		newWalletName := []byte(args[1])
+
+		wid, duplicate, err := client.FindWalletIDByName(walletName)
+
+		if wid == nil {
+			reportErrorf(errorCouldntFindWallet, string(walletName))
+		}
+
+		if err != nil {
+			reportErrorf(errorCouldntRenameWallet, err)
+		}
```

Comment:
I think you should check `err` first, so you get the more detailed error message.